keel-web/src/main/kotlin/com/netflix/spinnaker/keel/Main.kt

```diff
@@ -58,80 +44,7 @@ private val DEFAULT_PROPS = mapOf(
 @Import(PlatformComponents::class)
 @EnableAsync
 @EnableScheduling
-class KeelApplication {
-
-  private val log by lazy { LoggerFactory.getLogger(javaClass) }
-
-  @Autowired
-  lateinit var artifactRepository: ArtifactRepository
-
-  @Autowired
-  lateinit var resourceRepository: ResourceRepository
-
-  @Autowired
-  lateinit var deliveryConfigRepository: DeliveryConfigRepository
-
-  @Autowired(required = false)
-  var baseImageCache: BaseImageCache? = null
-
-  @Autowired
-  lateinit var instanceIdSupplier: InstanceIdSupplier
-
-  @Autowired(required = false)
-  var kinds: List<SupportedKind<*>> = emptyList()
-
-  @Autowired(required = false)
-  var resourceHandlers: List<ResourceHandler<*, *>> = emptyList()
-
-  @Autowired(required = false)
-  var constraintEvaluators: List<ConstraintEvaluator<*>> = emptyList()
-
-  @Autowired(required = false)
-  var artifactHandlers: List<ArtifactHandler> = emptyList()
-
-  @Autowired
-  lateinit var objectMappers: List<ObjectMapper>
-
-  @PostConstruct
-  fun registerResourceSpecSubtypes() {
-    resourceHandlers
-      .map { it.supportedKind }
-      .forEach { (kind, specClass) ->
-        log.info("Registering ResourceSpec sub-type {}: {}", kind, specClass.simpleName)
-        val namedType = NamedType(specClass, kind.toString())
-        objectMappers.forEach { it.registerSubtypes(namedType) }
-      }
-  }
-
-  @PostConstruct
-  fun registerConstraintSubtypes() {
-    constraintEvaluators
-      .map { it.supportedType }
-      .forEach { constraintType ->
-        log.info("Registering Constraint sub-type {}: {}", constraintType.name, constraintType.type.simpleName)
-        val namedType = NamedType(constraintType.type, constraintType.name)
-        objectMappers.forEach { it.registerSubtypes(namedType) }
-      }
-  }
-
-  @PostConstruct
-  fun initialStatus() {
-    sequenceOf(
-      ArtifactRepository::class to artifactRepository.javaClass,
-      ResourceRepository::class to resourceRepository.javaClass,
-      DeliveryConfigRepository::class to deliveryConfigRepository.javaClass,
-      BaseImageCache::class to baseImageCache?.javaClass,
-      InstanceIdSupplier::class to instanceIdSupplier.javaClass
-    )
-      .forEach { (type, implementation) ->
-        log.info("{} implementation: {}", type.simpleName, implementation?.simpleName)
-      }
-
-    log.info("Supporting resource kinds: {}", kinds.joinToString { it.kind.toString() })
-    log.info("Using resource handlers: {}", resourceHandlers.joinToString { it.name })
-    log.info("Using artifact handlers: {}", artifactHandlers.joinToString { it.name })
-  }
-}
+class KeelApplication
```

Comment:
That is one skinny class!

Reply:
I know, right?! But it pulls in all the rest... 🤷